README.md

```diff
@@ -1,23 +1,127 @@
+3. `subsidySpread`: The delta between the Base Rate and the Subsidy Rate. [RAY]
+
+## Functionality
+
+### `deposit`
+
+The `deposit` function is used to move funds from a given `ilk`'s `buffer` into the Conduit. From the Conduit, the funds are used to `supply` in the SparkLend Pool. The result is that:
+1. Funds are moved from the `buffer` to SparkLend's aToken for that asset.
+2. aTokens are minted and sent to the Conduit.
+3. The Conduit state to track the `ilk`'s portion of the aTokens in the Conduit is increased.
+
+<p align="center">
+  <img src="https://github.com/marsfoundation/spark-conduits/assets/44272939/2a7cf453-3a7b-4d04-a0cd-d390cfeb8ec2" height="500" />
+</p>
+
+### `withdraw`
+
+The `withdraw` function is used to `withdraw` funds from the SparkLend Pool into the Conduit. From the Conduit, the funds are sent to the `ilk`'s `buffer`. The result is that:
+
+1. Funds are moved from SparkLend's aToken for that asset to the `buffer`.
+2. The Conduit's aTokens corresponding to the underlying asset withdrawn are burned.
+3. The Conduit state to track the `ilk`'s portion of the aTokens in the Conduit is reduced.
+
+<p align="center">
+  <img src="https://github.com/marsfoundation/spark-conduits/assets/44272939/fd64b9e2-28f3-45c6-8deb-7d43283e9443" height="500" />
+</p>
+
+### `requestFunds`
+The `requestFunds` function is used to signal that a given `ilk` would like to withdraw funds from the Conduit. This is only possible when there is zero liquidity in SparkLend for the desired asset.
+
+The result is that:
+- The Conduit state to track the `ilk`'s requested funds in the Conduit is increased. Importantly, `totalRequestedShares[asset]` is increased.
+
+When `recompute()` is called in DaiInterestRateStrategy, it calls `getInterestData` on the SparkConduit. The `targetDebt` that is returned will be lower because it is `totalShares - totalRequestedShares`. This means that the `debtRatio` that is saved to storage will be greater than one, which means that the conditional logic to raise the interest rates (outlined in the `DaiInterestRateStrategy` section) is put into effect.
+
+`totalRequestedShares` only reduces when an `ilk` either:
+1. Cancels a withdrawal request.
+2. Withdraws funds from the Conduit.
+
+### `cancelRequest`
+The `cancelRequest` function is used to signal that a given `ilk` would like to cancel a request to withdraw funds from the Conduit. This is only possible when there is an active request to withdraw funds from the Conduit.
+
+The result is that:
+- The Conduit state to track the `ilk`'s requested funds in the Conduit is decreased. Importantly, `totalRequestedShares[asset]` is decreased.
+
+## Invariants
+
+$$ totalShares[asset] = \sum_{n=0}^{numIlks}{shares[asset][ilk]}  $$
+
+$$ totalRequestedShares[asset] = \sum_{n=0}^{numIlks}{requestedShares[asset][ilk]}  $$
+
+$$ totalShares[asset] = aToken.scaledBalanceOf(conduit) $$
+
+$$ getTotalDeposits(asset) = aToken.balanceOf(conduit) $$
```

Comment:
This is not strictly true due to permissionless transfer. `<=` holds though.

Reply:
Ah good point, I'll actually add notes about this

Reply:
Fixed